Lib/multiprocessing/resource_tracker.py

```diff
@@ -75,6 +75,20 @@ def _reentrant_call_error(self):
         raise ReentrantCallError(
             "Reentrant call into the multiprocessing resource tracker")
 
+    def __del__(self):
+        # making sure child processess are cleaned before ResourceTracker
+        # gets destructed.
+        # see https://github.com/python/cpython/issues/88887
+        try:
+            self._stop()
+        except AttributeError:
+            # AttributeError is likely caused by module teardown
+            # > __del__() can be executed during interpreter shutdown. As a
+            # > consequence, the global variables it needs to access (including
+            # > other modules) may already have been deleted or set to None.
+            # see https://docs.python.org/3/reference/datamodel.html#object.__del__
+            pass
+
     def _stop(self):
```

Comment:
You can keep references to os functions using this hack:

```suggestion
    def _stop(self, close=os.close, waitpid=os.waitpid,
              waitstatus_to_exitcode=os.waitstatus_to_exitcode):
```

Then in the function body, remove `os.` prefix.

Reply:
nice hack